### PR TITLE
Restructure-cli

### PR DESCRIPTION
**Pull request by PhilippeMoussalli**
With the introduction of a new compiler and runner for vertex, I feel the need to restructure the CLI a bit. This is a draft PR (docs and tests still need to be updated). 

**Running**

* Before

```bash
fondant run pipeline.py --local
```

* After

Mode is not longer a positional argument, Pipeline/reference is moved to the end (similar to docker). 

```bash
 fondant run local pipeline.py 
```
**Help**

* Before 

Arguments are not isolated per mode which makes it which argument belongs to which mode (e.g. extra volumes not needed for KFP)
```
❯ fondant run -h                 
usage: fondant run [-h] (--local | --kubeflow) [--output-path OUTPUT_PATH] [--extra-volumes EXTRA_VOLUMES [EXTRA_VOLUMES ...]] [--build-arg BUILD_ARG] [--host HOST] ref

positional arguments:
  ref                   Reference to the pipeline to run, can be a path to a spec file or a module containing the pipeline instance that will be compiled first (e.g. pipeline.py)

options:
  -h, --help            show this help message and exit
  --local
  --kubeflow
  --output-path OUTPUT_PATH, -o OUTPUT_PATH
                        Output directory
  --extra-volumes EXTRA_VOLUMES [EXTRA_VOLUMES ...]
                        Extra volumes to mount in containers
  --build-arg BUILD_ARG
                        Build arguments to pass to `docker build`. Format {key}={value}.
  --host HOST           KubeFlow pipeline host url
```
* After

```bash
❯fondant run -h  

positional arguments:
  {local,kubeflow}
    local           Local runner
    kubeflow        Kubeflow runner
  ref               Reference to the pipeline to run, can be a path to a spec file or a module containing the pipeline instance that will be compiled first (e.g. pipeline.py)

options:
  -h, --help        show this help message and exit
```

To further deep dive 

```bash
❯ fondant run local -h                  
usage: fondant run local [-h] [--output-path OUTPUT_PATH] [--extra-volumes EXTRA_VOLUMES [EXTRA_VOLUMES ...]] [--build-arg BUILD_ARG]

options:
  -h, --help            show this help message and exit
  --output-path OUTPUT_PATH, -o OUTPUT_PATH
                        Output path of compiled pipeline
  --extra-volumes EXTRA_VOLUMES [EXTRA_VOLUMES ...]
                        Extra volumes to mount in containers
  --build-arg BUILD_ARG
                        Build arguments for `docker build`
```

### CODE REVIEW

**Comment by RobbeSneyders** [comment]
Ok, this looks good to me @PhilippeMoussalli.

File: src/fondant/cli.py [hunk 1]
```diff
@@ -184,72 +170,85 @@ def register_compile(parent_parser):
...
+    )
+
+    # Local runner parser
+    local_parser.add_argument(
         "--output-path",
         "-o",
-        help="Output directory",
-        default=None,
+        help="Output path of compiled pipeline",
+        default="docker_compose.yml",
```
Comment (mrchtr) [inline src/fondant/cli.py:209]: Is there a reason for going away from the default docker-compose naming? 

```suggestion
        default="docker-compose.yml",
```

File: src/fondant/cli.py [hunk 2]
```diff
@@ -262,88 +261,108 @@ def register_run(parent_parser):
...
```
Comment (mrchtr) [inline src/fondant/cli.py:295]: See my comment above. 

```suggestion
        default="docker-compose.yml",
```

**Review by mrchtr — COMMENTED** [review]
Lgtm, just have one smaller suggestion.

File: src/fondant/cli.py (hunk 1 above)
Comment (PhilippeMoussalli) [reply src/fondant/cli.py:209]: Good catch, forgot that it was referenced like this

**Review by RobbeSneyders — APPROVED** [review]

**Pull request merged**